### DISCUSSION TITLE
is it possible to get the display used by xrdp-sesrun

### DISCUSSION THREAD

**Discussion by gnzsnz** [neutrinolabs/xrdp#3040] [Q&A] [answered]
I'm running a docker container with xfc (linuxserver rdesktop image). and I have created a script that will open a session using `xrdp-sesrun` once the session is open i need to lauch an application, which works well if the DISPLAY is set correctly.

usually i set DISPLAY=10 and works, but on the latest update of the image sometimes it's using 11 or 12. so i can not pre-define it.

is it possible to capture which one is using?

thanks in advance for your help and great software

```
root@04340439949c:/# xrdp -v
xrdp 0.9.17
  A Remote Desktop Protocol Server.
  Copyright (C) 2004-2020 Jay Sorg, Neutrino Labs, and all contributors.
  See https://github.com/neutrinolabs/xrdp for more information.

  Configure options:
      --enable-ipv6
      --enable-jpeg
      --enable-fuse
      --enable-rfxcodec
      --enable-opus
      --enable-painter
      --enable-vsock
      --build=x86_64-linux-gnu
      --prefix=/usr
      --includedir=${prefix}/include
      --mandir=${prefix}/share/man
      --infodir=${prefix}/share/info
      --sysconfdir=/etc
      --localstatedir=/var
      --disable-silent-rules
      --libdir=${prefix}/lib/x86_64-linux-gnu
      --libexecdir=${prefix}/lib/x86_64-linux-gnu
      --disable-maintainer-mode
      --disable-dependency-tracking
      --with-socketdir=/run/xrdp/sockdir
      build_alias=x86_64-linux-gnu
      CFLAGS=-g -O2 -ffile-prefix-map=/build/xrdp-asBei3/xrdp-0.9.17=. -flto=auto -ffat-lto-objects -flto=auto -ffat-lto-objects -fstack-protector-strong -Wformat -Werror=format-security
      LDFLAGS=-Wl,-Bsymbolic-functions -flto=auto -ffat-lto-objects -flto=auto -Wl,-z,relro -Wl,-z,now -Wl,--as-needed
      CPPFLAGS=-Wdate-time -D_FORTIFY_SOURCE=2 -Wno-error=deprecated-declarations
      PKG_CONFIG_PATH=/build/xrdp-asBei3/xrdp-0.9.17/pkgconfig

  Compiled with OpenSSL 3.0.2 15 Mar 2022
```

**Comment by gnzsnz** [comment]
on https://github.com/neutrinolabs/xrdp/issues/1898#issuecomment-849533599 @matt335672 implied that might be possible (?)

**Reply by matt335672** [reply]
I think xrdp-sesrun for that version of xrdp tells you what display it's used. Can you pick that up?

**Comment by gnzsnz** [comment]
thanks for pointing me on the right direction @matt335672 

if i do

```bash
echo "$(</run/secrets/password)" | xrdp-sesrun -F0 -s 127.0.0.1 testuser
[20240423-10:19:47] [WARN ] Using deprecated option '-s'
ok data=1 display=:10 GUID=98063e9769df1308631010e85f794230
```
I get the display in the otput, among warnings and other information

this is giving me the display

```bash
echo "$(</run/secrets/password)" | xrdp-sesrun -F0 -s 127.0.0.1 testuser| grep -e '^ok' | cut -d ' ' -f 3 | cut -d '=' -f 2
:10
```

if there is a better way please let me know. thanks in advance for your help

**Reply by matt335672** [reply]
You can use sed to cut down the number of processes you're using, but this is (IMO) a bit less readable. It's also quite fiddly to get the backslashes right:-

```
$ echo 'ok data=1 display=:10 GUID=98063e9769df1308631010e85f794230' | \
    sed -ne 's/^ok.*display=\(:[0-9]*\).*/\1/p'
:10

```

A breakdown:-
| feature | Explanation |
| ----- | ----- |
| sed -ne | Parse stdin with the following expression. Don't echo by default |
| \\(:[0-9]*\\)  | Match a colon followed by 0 or more digits, and place in variable \\1 |
| ^ok.\*display=\\(:[0-9]*\\).\* | Match a complete line starting with ok, with display=:[0-9]* in it. Store :[0-9]* in variable \\1 |
| s/&lt;regular-expression&gt;/\\1/p | Substitute the entire matched regular expression with \\1 and print the result if matched |

**Reply by gnzsnz** [reply, marked as answer]
This is fantastic @matt335672  thanks so much for your help.
